homeassistant/components/media_player/yamaha.py

```diff
@@ -0,0 +1,96 @@
+"""
+Support for Yamaha Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.yamaha/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice)
+from homeassistant.const import STATE_OFF, STATE_ON
+REQUIREMENTS = ['rxv==0.1.9']
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_YAMAHA = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yamaha platform."""
+    yamaha = YamahaDevice(config.get("name", "Yamaha Receiver"))
```

Comment:
This can now raise an exception if `find()` finds no receivers because you would request index `0` of an empty list. Please handle appropriately.